source/views/home/background.ts

```diff
@@ -0,0 +1,60 @@
+import * as ImagePicker from 'react-native-image-picker'
+import {
+	useMutation,
+	UseMutationResult,
+	useQuery,
+	UseQueryResult,
+} from '@tanstack/react-query'
+import * as storage from '../../lib/storage'
+import {queryClient} from '../../init/tanstack-query'
+
+export class ImagePickerError extends Error {}
+
+export const keys = {
+	background: ['settings', 'background'] as const,
+}
+
+async function chooseBackgroundImage(): Promise<string | undefined> {
+	const {assets, didCancel, errorCode, errorMessage} =
+		await ImagePicker.launchImageLibrary({mediaType: 'photo'})
+
+	if (errorCode || errorMessage) {
+		throw new ImagePickerError()
```

Comment:
Bubbling these errors would be good, I don't think we're doing that?

Reply:
IIRC if you use the system-provided file picker, the user doesn't have to provide access because you only get access to the photo(s) they select.